tcms/telemetry/static/telemetry/js/testing/test-case-health.js

```diff
@@ -0,0 +1,94 @@
+$(document).ready(() => {
+    loadInitialProduct();
+    loadInitialTestPlans();
+
+    document.getElementById('id_product').onchange = () => {
+        update_version_select_from_product();
+        update_build_select_from_product(true);
+        updateTestPlanSelectFromProduct(drawPage);
+    };
+
+    document.getElementById('id_version').onchange = drawPage;
+    document.getElementById('id_build').onchange = drawPage;
+    document.getElementById('id_test_plan').onchange = drawPage;
+
+    $('#id_after').on('dp.change', drawPage);
+    $('#id_before').on('dp.change', drawPage);
+
+    drawPage();
+})
+
+function drawPage() {
+    const query = {};
+
+    const productId = $('#id_product').val();
+    if (productId) {
+        query['case__plan__product'] = productId;
+    }
+
+    const versionId = $('#id_version').val();
+    if (versionId) {
+        query['case__plan__product_version'] = versionId;
+    }
+
+    const buildId = $('#id_build').val();
+    if (buildId) {
+        query['build_id'] = buildId;
+    }
+
+    const testPlanId = $('#id_test_plan').val();
+    if (testPlanId) {
+        query['case__plan__plan_id'] = testPlanId;
+    }
+
+    const dateBefore = $('#id_before');
+    if (dateBefore.val()) {
+        query['close_date__lte'] = dateBefore.data('DateTimePicker').date().format('YYYY-MM-DD 23:59:59');
+    }
+
+    const dateAfter = $('#id_after');
+    if (dateAfter.val()) {
+        query['close_date__gte'] = dateAfter.data('DateTimePicker').date().format('YYYY-MM-DD 00:00:00');
+    }
+
+    jsonRPC("Testing.test_case_health", query, data => {
+        drawTable('#test-case-health-table', data);
+    })
+}
+
+function drawTable(selector, data) {
+    $(`${selector} > tbody`).remove()
+
+    $(selector).DataTable({
+        data: data,
+        columns: [
+            {
+                data: null,
+                render: renderTestCaseColumn
+            },
+            {
+                data: null,
+                render: renderFailedExecutionsColumn
+            },
+            {
+                data: null,
+                render: renderPercentColumn
+            },
+        ],
+        paging: false,
+        ordering: false,
+        dom: "t"
+    });
+}
+
+function renderTestCaseColumn(data) {
```

Comment:
this could be extracted to `telemetry/utils.js`, but let's do that in a separate PR, which will include removing the duplicated logic as well